### PR TITLE
[dagster-tableau] Update docs with refreshable workbooks and materialization

### PR DESCRIPTION
**Pull request by maximearmstrong**
## Summary & Motivation

As title.

## How I Tested These Changes

Docs preview

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by maximearmstrong** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24864?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24864** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24864?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24862** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24862?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24826** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24826?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @maximearmstrong and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/content/integrations/tableau.mdx [hunk 1]
```diff
@@ -151,3 +155,34 @@ defs = Definitions.merge(
     marketing_team_workspace.build_defs(),
 )
 ```
+
+### Refresh and materialize Tableau assets
+
+You can use Dagster to orchestrate the refreshment of Tableau workbooks and materialization of Looker sheets and dashboards.
```
Comment (PedramNavid) [inline docs/content/integrations/tableau.mdx:161]:
```suggestion
You can use Dagster to refresh Tableau workbooks and materialize Looker sheets and dashboards.
```

File: docs/content/integrations/tableau.mdx [hunk 2]
```diff
@@ -151,3 +155,34 @@ defs = Definitions.merge(
...
+
+To refresh a Tableau workbook with Dagster, pass its definition by passing `StartWorkbookRefreshRequest` to `TableauCloudWorkspace.build_defs(...)` function.
```
Comment (PedramNavid) [inline docs/content/integrations/tableau.mdx:163]: I would skip this line and just go straight to the code, as the code will show this exact thing.

File: docs/content/integrations/tableau.mdx [hunk 3]
```diff
@@ -151,3 +155,34 @@ defs = Definitions.merge(
...
+
+Note that only workbooks created with extracts can be refreshed using this method. See more about [refreshing data sources](https://help.tableau.com/current/pro/desktop/en-us/refreshing_data.htm) in Tableau documentation website.
+
+When materializing the Tableau assets, Dagster will refresh the workbooks for which a `StartWorkbookRefreshRequest` has been created. Dagster will materialize the sheets and dashboards of workbooks that have been refreshed as `AssetMaterialization`, and the sheets and dashboards of the workbooks that have not been refreshed as `AssetObservation`.
```
Comment (PedramNavid) [inline docs/content/integrations/tableau.mdx:167]: This is a little confusing. Does a user need to know the distinction here? I think I would remove this whole paragraph. The API docs for `StartWorkbookRefreshRequest` should outline what this function does, but I don't think the guide needs it.

**Review by PedramNavid — CHANGES_REQUESTED** [review]

File: docs/content/integrations/tableau.mdx (hunk 3 above)
Comment (maximearmstrong) [reply docs/content/integrations/tableau.mdx]: That makes sense. All changes requested done in [b954994](https://github.com/dagster-io/dagster/pull/24864/commits/b954994282ac767eee6f80278413b0fe41f43e52)

**Comment by cmpadden** [comment]
Merged changes from #25369 to keep Graphite stack intact. Confirmed unit tests are running green:
```
============================================================= test session starts ==============================================================
platform darwin -- Python 3.11.7, pytest-8.3.2, pluggy-1.5.0
rootdir: /Users/colton/src/dagster
configfile: pyproject.toml
plugins: syrupy-4.6.1, cov-5.0.0, time-machine-2.15.0, cases-3.8.5, buildkite-test-collector-0.1.8, rerunfailures-14.0, typeguard-4.3.0, anyio-4.4.0, mock-3.14.0, hypothesis-6.111.0, xdist-3.6.1, requests-mock-1.12.1
collected 15 items

dagster_tableau_tests/test_asset_specs.py ....                                                                                           [ 26%]
dagster_tableau_tests/test_pending_repo.py ..                                                                                            [ 40%]
dagster_tableau_tests/test_resources.py ....                                                                                             [ 66%]
dagster_tableau_tests/test_translator.py ....                                                                                            [ 93%]
dagster_tableau_tests/test_version.py .                                                                                                  [100%]

============================================================= 15 passed in 13.11s ==============================================================
```

**Review by C00ldudeNoonan — APPROVED** [review]
Looks great, the api references will help users understand what they need to do.

**Pull request merged**